src/processors/universal.ts

```diff
@@ -765,10 +765,13 @@ class UniversalProcessor extends BaseOrderProcessor {
       if (new BigNumber(requiredReserveDstAmount).isEqualTo('0')) {
         message = 'not enough give amount to cover operating expenses';
       } else {
-        const takeAmountDrop = new BigNumber(profitableTakeAmount)
-          .multipliedBy(100)
-          .div(orderInfo.order.take.amount.toString());
-        const takeAmountDropShare = BigNumber(100).minus(takeAmountDrop).toFixed(2);
+        let takeAmountDropShare: string = '100';
+        if (orderInfo.order.take.amount !== 0n) {
+          const takeAmountDrop = new BigNumber(profitableTakeAmount)
+            .multipliedBy(100)
+            .div(orderInfo.order.take.amount.toString());
+          takeAmountDropShare = BigNumber(100).minus(takeAmountDrop).toFixed(2);
+        }
```

Comment:
Why do we need this change? 

Shouldn't `requiredReserveDstAmount` be equal to zero is order is not profitable? Please provide a use case for this change